PROTO_tests/tests/random_test.py

```diff
@@ -47,70 +52,75 @@ def test_integers(self):
+
         # verify same seed gives reproducible arrays
+
         rng = ak.random.default_rng(18)
+        rnfunc = rng.integers if data_type is ak.int64 else rng.uniform
+        pda = rnfunc(-(2**32),2**32,10)
+        pda_copy = pda[:]
+        rng.shuffle(pda)
 
-        int_pda = rng.integers(-(2**32), 2**32, 10)
-        pda_copy = int_pda[:]
-        # shuffle int_pda in place
-        rng.shuffle(int_pda)
-        # verify all the same elements are in permutation as the original
-        assert (ak.sort(int_pda) == ak.sort(pda_copy)).all()
+        assert check (ak.sort(pda),ak.sort(pda_copy),data_type)
 
-        float_pda = rng.uniform(-(2**32), 2**32, 10)
-        pda_copy = float_pda[:]
-        rng.shuffle(float_pda)
-        # verify all the same elements are in permutation as the original
-        assert (ak.sort(float_pda) == ak.sort(pda_copy)).all()
+        # verify all the same elements are in the permutation as in the original
 
         rng = ak.random.default_rng(18)
+        rnfunc = rng.integers if data_type is ak.int64 else rng.uniform
+        pda_prime = rnfunc(-(2**32),2**32,10)
+        rng.shuffle(pda_prime)
 
-        pda = rng.integers(-(2**32), 2**32, 10)
-        rng.shuffle(pda)
-        assert (pda == int_pda).all()
+        assert check (pda,pda_prime,data_type)
 
-        pda = rng.uniform(-(2**32), 2**32, 10)
-        rng.shuffle(pda)
-        assert np.allclose(pda.to_list(), float_pda.to_list())
+    @pytest.mark.parametrize("data_type",INT_FLOAT)
+    def test_permutation(self,data_type) :
+
+        # ints are checked for equality; floats are checked for closeness
+
+        check = lambda a,b,t : (ak.sort(a)==ak.sort(b)).all() if t is ak.int64 else \
+                np.allclose(a.to_list(),b.to_list())
 
-    def test_permutation(self):
-        # verify same seed gives reproducible arrays
         rng = ak.random.default_rng(18)
-        # providing just a number permutes the range(num)
         range_permute = rng.permutation(20)
-        assert (ak.arange(20) == ak.sort(range_permute)).all()
+        assert (ak.arange(20) == ak.sort(range_permute)).all() # range is always int
 
-        pda = rng.integers(-(2**32), 2**32, 10)
-        array_permute = rng.permutation(pda)
-        # verify all the same elements are in permutation as the original
-        assert (ak.sort(pda) == ak.sort(array_permute)).all()
+        # verify same seed gives reproducible arrays
 
-        pda = rng.uniform(-(2**32), 2**32, 10)
-        float_array_permute = rng.permutation(pda)
-        # verify all the same elements are in permutation as the original
-        assert np.allclose(ak.sort(pda).to_list(), ak.sort(float_array_permute).to_list())
+        rng = ak.random.default_rng(18)
+        rnfunc = rng.integers if data_type is ak.int64 else rng.uniform
+        pda = rnfunc(-(2**32),2**32,10)
+        permuted = rng.permutation(pda)
+        assert check(ak.sort(pda),ak.sort(permuted),data_type)
```

Comment:
The check immediately below that is to test whether using the same seed twice gives the same permutation.  Sorting within the check would fail to check that.

Reply:
In fact, the more I looked at this, the more I realized I didn't want any sort in check (I was sorting if the data type was ak.int64), so I removed that.